packages/sources/coingecko/src/util.ts

```diff
@@ -1,29 +1,31 @@
-import { Config } from '@chainlink/types'
-import { Requester } from '@chainlink/ea-bootstrap'
+import { makeExecute } from './adapter'
+import { executeSync } from '@chainlink/ea-bootstrap'
+import { CoinsResponse } from './endpoint/coins'
 
-type CoinsListResponse = {
-  id: string
-  symbol: string
-  name: string
-}[]
-
-const customError = (data: any) => {
-  if (Object.keys(data).length === 0) return true
-  return false
-}
-
-export const getCoinIds = async (config: Config): Promise<CoinsListResponse> => {
-  const url = '/coins/list'
-  const options = {
-    ...config.api,
-    url,
-  }
-  return (await Requester.request(options, customError)).data as CoinsListResponse
+export const getCoinIds = (id: string): Promise<CoinsResponse> => {
```

Comment:
Just threw these in the 2 EAs that needed them, but we could probably abstract this to be a utility method exposed by `ea-bootstrap` where `execute` and and `options` are passed in as arguments.